shared/util_linux.go

```diff
@@ -384,3 +386,90 @@ func SetSize(fd int, width int, height int) (err error) {
 	}
 	return nil
 }
+
+// This uses ssize_t llistxattr(const char *path, char *list, size_t size); to
+// handle symbolic links (should it in the future be possible to set extended
+// attributed on symlinks): If path is a symbolic link the extended attributes
+// associated with the link itself are retrieved.
+func llistxattr(path string, list []byte) (sz int, err error) {
+	var _p0 *byte
+	_p0, err = unix.BytePtrFromString(path)
+	if err != nil {
+		return
+	}
+	var _p1 unsafe.Pointer
+	if len(list) > 0 {
+		_p1 = unsafe.Pointer(&list[0])
+	} else {
+		_p1 = unsafe.Pointer(nil)
+	}
+	r0, _, e1 := unix.Syscall(unix.SYS_LLISTXATTR, uintptr(unsafe.Pointer(_p0)), uintptr(_p1), uintptr(len(list)))
+	sz = int(r0)
+	if e1 != 0 {
```

Comment:
The type is `syscall.Errno` and `gc` complains for me when you compare `syscall.Errno` against `nil`. :)